code/game/objects/items/melee/baton.dm

```diff
@@ -984,3 +984,51 @@
 	else
 		stuff_in_hand.forceMove(user.drop_location())
 		stuff_in_hand.loc.visible_message(span_warning("[stuff_in_hand] suddenly appears!"))
+
+
+/obj/item/melee/baton/nunchaku
+	name = "Syndie Fitness Nunchuks"
+	desc = "The most common fitness equipment in the entire syndicate, titanium rods weigh strictly 13 pounds"
+	desc_controls = "Left click to stun, right click to harm. Throw mode counterattack any melee/throwable attacks."
+	icon_state = "nunchaku"
+	lefthand_file = 'icons/mob/inhands/weapons/melee_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/melee_righthand.dmi'
+	inhand_icon_state = "nunchaku"
+	worn_icon_state = "nunchaku"
+	attack_verb_continuous = list("beats", "whips", "smashes", "punishes")
+	attack_verb_simple = list("beat", "whip", "smash", "punish")
+	hitsound = 'sound/items/weapons/chainhit.ogg'
+	block_sound = 'sound/items/weapons/block_shield.ogg'
+	slot_flags = ITEM_SLOT_BELT
+	cooldown = CLICK_CD_MELEE
+	knockdown_time = 250 MILLISECONDS
+	demolition_mod = 1.5
+	stamina_damage = 35 // 3 hit stamcrit
+	stun_armour_penetration = 30 // bronze-silver telescopic
+	force = 16 // 7 hit crit
+	bare_wound_bonus = 5 // 5 hit crit
+
+/obj/item/melee/baton/nunchaku/Initialize(mapload)
+	. = ..()
+	// change icon state after throw/attacks
+	RegisterSignals(src, list(COMSIG_MOVABLE_POST_THROW, COMSIG_ITEM_AFTERATTACK), PROC_REF(randomize_state))
+
+/obj/item/melee/baton/nunchaku/proc/randomize_state()
+	icon_state = pick(list("nunchaku", "nunchaku_x", "nunchaku_y"))
+	update_icon_state()
```

Comment:
Rather than hooking these signals, just override the reespective procs `throw_at` and `afterattack`

Reply:
Also, that `update_icon_state` call isn't doing anything